appeals/web/src/server/appeals/appeal-details/lpa-questionnaire/changed-listed-buildings/changed-listed-buildings.controller.js

```diff
@@ -0,0 +1,371 @@
+import logger from '#lib/logger.js';
+import { addNotificationBannerToSession } from '#lib/session-utilities.js';
+import { getLpaQuestionnaireFromId } from '../lpa-questionnaire.service.js';
+import {
+	addChangedListedBuildingCheckAndConfirmPage,
+	addChangedListedBuildingPage,
+	changeChangedListedBuildingCheckAndConfirmPage,
+	changeChangedListedBuildingPage,
+	manageChangedListedBuildingPage,
+	removeChangedListedBuildingPage
+} from './changed-listed-buildings.mapper.js';
+import {
+	addChangedListedBuilding,
+	removeChangedListedBuilding,
+	changeChangedListedBuilding
+} from './changed-listed-buildings.service.js';
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const getAddChangedListedBuilding = async (request, response) => {
+	return renderAddChangedListedBuilding(request, response);
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderAddChangedListedBuilding = async (request, response) => {
+	const { errors, currentAppeal } = request;
+
+	const mappedPageContents = addChangedListedBuildingPage(
+		currentAppeal,
+		request.session.affectedListedBuilding
+	);
+
+	return response
+		.status(200)
+		.render('patterns/change-page.pattern.njk', { pageContent: mappedPageContents, errors });
+};
+
+/**
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+export const postAddChangedListedBuilding = async (request, response) => {
+	request.session.affectedListedBuilding = request.body['changedListedBuilding'];
+	try {
+		if (request.errors) {
+			return renderAddChangedListedBuilding(request, response);
+		}
+
+		const {
+			params: { appealId, lpaQuestionnaireId }
+		} = request;
+
+		return response.redirect(
+			`/appeals-service/appeal-details/${appealId}/lpa-questionnaire/${lpaQuestionnaireId}/changed-listed-buildings/add/check-and-confirm`
+		);
+	} catch (error) {
+		logger.error(error);
+	}
+
+	delete request.session.affectedListedBuilding;
+	return response.status(500).render('app/500.njk');
```

Comment:
Nothing throws here so no need for the try-catch :)

Also there's a middleware at the bottom of `appeals/web/src/server/app/app.express.js` that automatically renders the error pages if errors are thrown, so no need to render it manually. You can safely let errors bubble up.

```suggestion
	if (request.errors) {
		return renderAddChangedListedBuilding(request, response);
	}

	const {
		params: { appealId, lpaQuestionnaireId }
	} = request;

	delete request.session.affectedListedBuilding;

	return response.redirect(
		`/appeals-service/appeal-details/${appealId}/lpa-questionnaire/${lpaQuestionnaireId}/changed-listed-buildings/add/check-and-confirm`
	);
```